backend/app/libs/tokenlists.py

```diff
@@ -0,0 +1,71 @@
+from asyncio import gather, sleep
+from json.decoder import JSONDecodeError
+from typing import Coroutine
+
+from celery.utils.log import get_task_logger
+from httpx import AsyncClient, HTTPStatusError, RequestError, Timeout
+
+from .storage_helpers import retrieve_token_whitelist, store_token_whitelist
+
+
+async def retry(method: Coroutine, url: str, await_time: float):
+    await sleep(await_time)
+    await method(url)
+
+
+async def get_token_list(token_list: str):
+    timeout = Timeout(10.0, read=30.0, connect=15.0)
+    logger = get_task_logger(__name__)
+    async with AsyncClient(timeout=timeout) as client:
+        try:
+            resp = await client.get(token_list)
+            resp.raise_for_status()
+        except HTTPStatusError:
+            if resp.status_code == 404:
+                logger.warn("404 response for %s, aborting...", token_list)
+                return
+            logger.error("error receiving token list for %s, retrying...", token_list)
+            await retry(client.get, token_list, 10.0)
```

Comment:
I'm not sure this stores the result anywhere…

It would seem reasonable to me to move this retrying logic to a separate function (could be a decorator), that we can test.

But first, do we really need it for now by the way? It seems no, as we're talking about querying results repeatedly in a scheduler, and missing one answer from time to time doesn't seem to matter much.

Reply:
You're right, a bit of oversight with not storing the result of the retry 😅.

> But first, do we really need it for now by the way? It seems no, as we're talking about querying results repeatedly in a scheduler, and missing one answer from time to time doesn't seem to matter much.

Seems like in that case it just makes sense to stop execution of the task, which I'm in favour of.